ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ethernet_interfaces.schema.yml

```
@@ -0,0 +1,785 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  ethernet_interfaces:
+    type: list
+    primary_key: name
+    convert_types:
+    - dict
+    description: Ethernet interfaces
+    items:
+      type: dict
+      keys:
+        name:
+          type: str
+          required: true
+        peer:
+          type: str
+        peer_interface:
+          type: str
+        peer_type:
+          type: str
+        description:
+          type: str
+        shutdown:
+          type: bool
+        speed:
+          type: str
+          description: Speed can be interface_speed or forced interface_speed or auto interface_speed
+        mtu:
+          type: int
+          convert_types:
+          - str
+        l2_mtu:
+          type: int
+          convert_types:
+          - str
+          description: If l2_mtu defined this profile should only be used for platforms supporting the "l2 mtu" CLI
+        vlans:
+          type: str
+          convert_types:
+          - int
+          description: List of vlans as string
+        native_vlan:
+          type: int
+          convert_types:
+          - str
+        native_vlan_tag:
+          type: bool
+          default: false
+        mode:
+          type: str
+          valid_values: ["access", "dot1q-tunnel", "trunk", "trunk phone"]
+        phone:
+          type: dict
+          keys:
+            trunk:
+              type: str
+              valid_values: ["tagged", "untagged"]
+            vlan:
+              type: int
+              min: 1
+              max: 4094
+              convert_types:
+              - str
+        l2_protocol:
+          type: dict
+          keys:
+            encapsulation_dot1q_vlan:
+              type: int
+              convert_types:
+              - str
+              description: Vlan tag to configure on sub-interface
+        trunk_groups:
+          type: list
+          items:
+            type: str
+        type:
+          type: str
+          valid_values: ["routed", "switched", "l3dot1q", "l2dot1q"]
+          description: l3dot1q and l2dot1q are used for sub-interfaces. The parent interface should be defined as routed.
+        snmp_trap_link_change:
+          type: bool
+          description: SNMP trap link change
+        flowcontrol:
+          type: dict
+          keys:
+            received:
+              type: str
+              valid_values: ["received", "send", "on"]
```

There are some issues with the existing docs.
```suggestion
              valid_values: ["desired", "on", "off"]
```